Sources/Fuzzilli/Lifting/JavaScriptLifter.swift

```diff
@@ -886,7 +886,7 @@ public class JavaScriptLifter: Lifter {
                 w.assign(expr, to: instr.output)
 
             case .loadNamedVariable(let op):
-                w.assign(Identifier.new(op.variableName), to: instr.output)
+                w.assign(Identifier.new(op.variableName), to: instr.output, forceInlining: true)
```

Comment:
I will try to justify this with an example. Let's pretend the newly added forceInlining parameter doesn't exist. Then we would have an issue in the part where we lift FuzzIL->JS 

## JS before compiling to FuzzIL
```javascript
for (a of ["new a"]) {}

b = "old b";
for (b of ["new b"]) {}
```
## JS after lifting FuzzIL
```javascript
const v1 = ["new a"];
let v2 = a;
for (v2 of v1) {
}

b = "old b";
const v5 = ["new b"];
let v6 = b;
for (v6 of v5) {
}
```

As we can see here, the lifting was not successful because the iterator in the loops became a new js variable even though we want to use the global variable. The reason why inlining can not happen here is a bit specific: The For*LoopWithReassignment loops we defined in **JSOperations.swift** have the property **reassigning** defined in **Semantics.swift**. This causes **shouldTryInlining** to return false. This is why I introduced forceInlining.

Reply:
Right, that's what I thought. I *think* we should be able to handle this differently though: can you try using `w.declare(instr.output, as: op.variableName)` and see if that works, too?

More high-level, my (small) concern with this is that the lifted JS doesn't really fit the FuzzIL code, because there we have a `LoadNamedVariable`, which does basically create a new local variable with the content of a named variable. So it's somewhat unintuitive that later assignments to that new local variable also change the named variable. However, after thinking about this for a bit, I think what we should actually change is the way named variables work in FuzzIL. That should also allow us to (eventually) fix a number of other correctness issues with the compiler. What I would suggest is replacing `LoadNamedVariable`, `StoreNamedVariable`, and `DefineNamedVariable` with a single operation:
```
v3 <- CreateNamedVariable "foo", $declarationMode, [initialValue]
```
Where the declaration mode is something like
```
enum NamedVariableDeclarationMode {
    case none
    case global
    case var
    case let
}
```
And so reflects the different types of JS variables. Then, the lifter will simply lift this variable to its name (and potentially create a `var` or `let` declaration statement), and there is no local variable anymore (which is what we want here and elsewhere). For your example above, we would simply compile it to
```
v2 <- LoadString "old b"
v3 <- CreateNamedVariable "b", .global, v2
ForInLoopWithReassign v1, v3
    ...
EndForInLoop
```

What do you think about this? We'd certainly want to do this in a follow-up PR, but I think it would be good to do this somewhat soon-ish so that we can see if it works out and because it might be important for fixing a number of other correctness issues in the compiler (basically for every program where the names of variables are important, e.g. because they are referenced from eval'd code, inside a `with`, used in an object literal, etc.).

Reply:
Ok, I gave that idea a shot now: https://github.com/googleprojectzero/fuzzilli/pull/486 let me know what you think and if that should solve these issues here. No rush though :)